katacomb/katacomb/util/image_util.py

```diff
@@ -1,215 +1,94 @@
+    ax.set_ylim(-0.5 + y_trim, image_height - y_trim - 0.5)
+    return fig, ax
+
+
+def _plot(data, bunit, caption, ax, extent, vmin, vmax):
+    if bunit == 'JY/BEAM':
+        # This is not FITS-standard, but is AIPS standard and is output by AIPS generated FITS images
+        # as well as older versions of katsdpimager
+        unit = units.Jy / units.beam
+    else:
+        unit = units.Unit(bunit)
+    data <<= unit
+    vmin <<= unit
+    vmax <<= unit
+    # If the flux is low, use µJy/beam or mJy/beam to keep scale sane
+    if vmax < 100 * (units.uJy / units.beam):
+        data = data.to(units.uJy / units.beam)
+    elif vmax < 100 * (units.mJy / units.beam):
+        data = data.to(units.mJy / units.beam)
+    vmin = vmin.to(data.unit)
+    vmax = vmax.to(data.unit)
+    if not ax.images:
+        im = ax.imshow(data.value, origin='lower', cmap='afmhot', aspect='equal',
+                       vmin=vmin.value, vmax=vmax.value, extent=extent)
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes('right', pad='3%', size='5%', axes_class=matplotlib.axes.Axes)
+        cbar = ax.get_figure().colorbar(im, cax=cax, orientation='vertical')
+        # simply using data.unit.to_string('unicode') draws an ASCII-art
+        # fraction, which doesn't end up looking very good. But we want
+        # unicode to properly render µJy (rather than uJy).
+        unit_label = (data.unit * units.beam).to_string('unicode') + ' / beam'
+        cbar.set_label(unit_label)
+    else:
+        im = ax.images[0]
+        im.set_data(data)
+        im.set_extent(extent)
+
+    if caption:
+        ax.set_title(f'{caption}')
+
+
+def write_image(input_file, output_file, width=1024, height=768, dpi=DEFAULT_DPI,
+                slices=('x', 'y', 0, 0), trim=0.125, caption=None):
+    """Write an image plane to a file from a single FITS file.
 
     Parameters
     ----------
-    image : :class:`ImageFacade`
-        The image to plot
-    filename : str
-        Disk location of output. Filename extension determins
-        format of the output as for `matplotlib.pylab.savefig`
-    plane : int
-        The image plane to plot
-    drop_edge : float
-        Fraction of pixels on each edge of the image to clip.
-    cmap : str or :class:`matplotlib.colors.Colormap`
-        Matplotlib style colormap to use (passed to `imshow`)
-    display_size : float
-        Size in inches for the `matplotlib.pylab.figure` instance.
+    input_file : str
+        Source FITS file
+    output_file : str
+        Output image file, including extension
+    width, height : int
+        Dimensions of output image
     dpi : int
-        Dots per inch desired for figure.
-        (Passed to `matplotlib.pylab.savefig`)
+        DPI of output image
+    slices : list
+        Choice of image dimensions. Passed to :class:`WCSAxes`
+    trim : float
+        Fraction of pixels on each edge of the image to clip
```

Comment:
Have you considered automatically trimming to the bounding box of non-NaN pixels?

Reply:
That's a good idea.

I'll have to think about / experiment with the best way to do it.

Reply:
I imagine something involving np.min/np.max, np.nonzero and np.isfinite should be able to give axis bounds fairly easily. It will just need some corner case handling in case an image is all-NaN for some reason.